console/src/test/java/org/georchestra/console/ws/backoffice/orgs/OrgsControllerTest.java

```diff
@@ -1,33 +1,137 @@
+import org.georchestra.console.model.DelegationEntry;
+import org.georchestra.console.ws.utils.Validation;
 import org.json.JSONException;
 import org.json.JSONObject;
+import org.junit.Before;
 import org.junit.Test;
 import org.mockito.Mockito;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken;
 
+import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
+import java.io.PrintWriter;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.mockito.AdditionalMatchers.aryEq;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
 
 public class OrgsControllerTest {
 
+    private DelegationEntry mockEntry1;
+    private DelegationEntry mockEntry2;
+    private Org mockOrg;
+    private OrgExt mockOrgExt;
+    private OrgsDao mockOrgsDao;
+    private DelegationDao delegationDaoMock;
+
+    @Before
+    public void grantRight()
+    {
+        // Set user connected through spring security
+        List<GrantedAuthority> role = new LinkedList<GrantedAuthority>();
+        role.add(new SimpleGrantedAuthority("ROLE_SUPERUSER"));
+        Authentication auth = new PreAuthenticatedAuthenticationToken("testadmin",
+                null,
+                role);
+        SecurityContextHolder.getContext().setAuthentication(auth);
+    }
+
     @Test
     public void updateFromRequestTakePendingIntoAccount() throws JSONException, IOException {
-        OrgsDao mockOrgsDao = Mockito.mock(OrgsDao.class);
+        OrgsDao mockOrgsDao = mock(OrgsDao.class);
         OrgsController toTest = new OrgsController(mockOrgsDao);
-        Org mockOrg = Mockito.mock(Org.class);
+        Org mockOrg = mock(Org.class);
         JSONObject jsonInput = new JSONObject();
         jsonInput.put("pending", true);
 
         toTest.updateFromRequest(mockOrg, jsonInput);
 
-        Mockito.verify(mockOrg).setPending(true);
+        verify(mockOrg).setPending(true);
 
         jsonInput = new JSONObject();
         jsonInput.put("pending", false);
 
         toTest.updateFromRequest(mockOrg, jsonInput);
 
-        Mockito.verify(mockOrg).setPending(false);
+        verify(mockOrg).setPending(false);
+    }
+
+    @Test
+    public void deleteOrgDeleteDelegation() throws IOException, SQLException {
+        OrgsController toTest = createToTest();
+        HttpServletResponse mockResponse = mock(HttpServletResponse.class);
```

Comment:
spring-mock also provides a MockHttpServletRequest (https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/mock/web/MockHttpServletRequest.html), but that is ok